corehq/apps/integration/payments/services.py

```diff
@@ -99,5 +98,17 @@ def _get_payee_details(case_data: dict) -> PartyDetails:
         raise PaymentRequestError("Invalid payee details")
 
 
-def _payment_is_verified(payee_case: CommCareCase):
-    return payee_case.case_json[PaymentProperties.PAYMENT_VERIFIED] == 'True'
+def _validate_payment_request(payee_case: CommCareCase):
+    case_data = payee_case.case_json
+    if not _payment_is_verified(case_data):
+        raise PaymentRequestError("Payment has not been verified")
+    if _payment_already_submitted(case_data):
+        raise PaymentRequestError("Payment has already been submitted")
+
+
+def _payment_is_verified(case_data: dict):
+    return case_data.get(PaymentProperties.PAYMENT_VERIFIED, '') == 'True'
```

Comment:
Big nit: The default of `None` can simply be used here since that will also cause this check to fail:
```suggestion
    return case_data.get(PaymentProperties.PAYMENT_VERIFIED) == 'True'
```